typings/internal-ambient.d.ts

```diff
@@ -215,3 +215,71 @@ interface ResizeObserverEntry {
    */
   readonly contentRect: DOMRectReadOnly;
 }
+
+interface WeakRef<T extends object> {
+  readonly [Symbol.toStringTag]: "WeakRef";
+
+  /**
+   * Returns the WeakRef instance's target object, or undefined if the target object has been
+   * reclaimed.
+   */
+  deref(): T | undefined;
+}
+
+interface WeakRefConstructor {
+  readonly prototype: WeakRef<any>;
+
+  /**
+   * Creates a WeakRef instance for the given target object.
+   * @param target The target object for the WeakRef instance.
+   */
+  new<T extends object>(target?: T): WeakRef<T>;
+}
+
+declare var WeakRef: WeakRefConstructor;
+
+interface FinalizationRegistry {
+  readonly [Symbol.toStringTag]: "FinalizationRegistry";
+
+  /**
+   * Registers an object with the registry.
+   * @param target The target object to register.
+   * @param heldValue The value to pass to the finalizer for this object. This cannot be the
+   * target object.
+   * @param unregisterToken The token to pass to the unregister method to unregister the target
+   * object. If provided (and not undefined), this must be an object. If not provided, the target
+   * cannot be unregistered.
+   */
+  register(target: object, heldValue: any, unregisterToken?: object): void;
+
+  /**
+   * Unregisters an object from the registry.
+   * @param unregisterToken The token that was used as the unregisterToken argument when calling
+   * register to register the target object.
+   */
+  unregister(unregisterToken: object): void;
+
+  /**
+   * Triggers callbacks for an implementation-chosen number of objects in the registry that have
+   * been reclaimed but whose callbacks have not yet been called.
+   *
+   * Note: The number of entries for reclaimed objects that are cleaned up from the registry
+   * (calling the cleanup callbacks) is implementation-defined. An implementation might remove
+   * just one eligible entry, or all eligible entries, or somewhere in between.
+   * @param callback If given, the registry uses the given callback instead of the one it was
+   * created with.
+   */
+  cleanupSome?(callback?: (heldValue: any) => void): void;
+}
+
+interface FinalizationRegistryConstructor {
+  readonly prototype: FinalizationRegistry;
+
+  /**
+   * Creates a finalization registry with an associated cleanup callback
+   * @param cleanupCallback The callback to call after an object in the registry has been reclaimed.
+   */
+  new(cleanupCallback: (heldValue: any) => void): FinalizationRegistry;
+}
+
+declare var FinalizationRegistry: FinalizationRegistryConstructor;
```

Comment:
Is this copy/pasted from somewhere? Perhaps note where?

Reply:
it's from https://github.com/microsoft/TypeScript/pull/38232

Reply:
@nornagon do you know why am I getting this linter error even though `WeakRef` is declared?
```
/home/builduser/project/src/electron/spec-main/api-browser-window-spec.ts
  74:23  error  'WeakRef' is not defined  no-undef

/home/builduser/project/src/electron/spec-main/api-menu-spec.ts
  845:23  error  'WeakRef' is not defined  no-undef
```

Reply:
No, I'm not sure why that would be. Seems like `declare var WeakRef` on L239 should fix that?